core/src/main/java/org/jboss/galleon/util/IoUtils.java

```diff
@@ -164,13 +169,24 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
                              if (!Files.isDirectory(targetDir)) {
                                  throw e;
                              }
+                        } catch (AccessDeniedException e) {
+                            if (!skipExistingFiles && Files.exists(targetDir)) {
+                                throw e;
+                            }
                         }
                         return FileVisitResult.CONTINUE;
                     }
                     @Override
                     public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
                         throws IOException {
-                        Files.copy(file, target.resolve(source.relativize(file).toString()), StandardCopyOption.REPLACE_EXISTING);
+                        final Path targetFile = target.resolve(source.relativize(file).toString());
+                        try {
+                            Files.copy(file, targetFile, StandardCopyOption.REPLACE_EXISTING);
+                        } catch (AccessDeniedException e) {
+                            if (!skipExistingFiles && Files.exists(targetFile)) {
```

Comment:
Do we need the Files.exists check? If skipExistingFiles is false and the file doesn't exit, the error is ignored. That change the original behavior.

Reply:
You're right that should be the other way round - raise the exception if the skip is true but the file doesn't exist